### ISSUE TITLE
StringConcatenation vs. RedundantInterpolation for multiline strings

### ISSUE THREAD

**Issue by chris-hewitt** [rubocop/rubocop#12510]
Consider the following example. The first line passes. But in the second case, when we have to break into a multiline string and the interpolation is at the end, it seems that `Style/StringConcatenation` always takes precedence over `Style/RedundantInterpolation`.

```ruby
                                           | max line length
puts "This text ends with #{some_method}"  |
                                           | 
puts "This text ends with #{some_much_longer_method}"
                                           | 
```

**Expected passing syntax: (violates Style/StringConcatenation)**
```ruby
                                           | max line length
puts "This text ends with #{some_method}"  |
                                           | 
puts 'This text ends with ' +              |
     some_much_longer_method               |
                                           | 
```

**Actual passing syntax:**
```ruby
                                           | max line length
puts "This text ends with #{some_method}"  |
                                           | 
puts 'This text ends with ' \              |
     "#{some_much_longer_method}"          |
                                           | 
```

**Version:**
```shell
$ bundle exec rubocop -V
1.58.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 2.7.6) [x86_64-linux]
```

**Comment by chris-hewitt** [comment]
This applies anywhere an interpolated string segment would appear on its own line. Here's a complex example:
```ruby
puts 'The ' +                          | max line length
     domesticated_animal.long_name +   |
     ' is known for its beautiful ' +  |
     domesticated_animal.feature +     |
     " throughout #{region}."          |
```
(I'm not even sure how I would write this one; maybe it's not worth thinking about right now)

**Comment by chris-hewitt** [comment]
I should mention that autocorrect may not want to change, since it doesn't know if the interpolated segment needs to be cast to a string. But when the developer intentionally writes `puts 'text ' + ` `     code_returning_text`, I think the cop should be ok with it.

**Comment by kellyky** [comment]
Hi, I was able to reproduce this as well using the example in the description (though not for the more [complex example](https://github.com/rubocop/rubocop/issues/12510#issuecomment-1839307714)):

If I'm correctly understanding the documentation and the issue, it seems this is not the expected behavior for `Style/StringConcatenation`. 

Sharing my version as well:

```
$ bundle exec rubocop -V
1.58.0 (using Parser 3.3.5.0, rubocop-ast 1.32.3, running on ruby 3.3.0) [arm64-darwin23]
```

According to the documentation,  `Style/StringConcatenation` should not register an offense for string concatenation across multiple lines, which appears to be the reported case. However, it does register an offense. (The documentation also indicates that if LineEndConcatenation is enabled, it should catch this instead, not Style/StringConcatenation.)
- https://docs.rubocop.org/rubocop/cops_style.html#stylestringconcatenation.

**Comment by dvandersluis** [comment]
> According to the documentation, `Style/StringConcatenation` should not register an offense for string concatenation across multiple lines, which appears to be the reported case. However, it does register an offense. (The documentation also indicates that if LineEndConcatenation is enabled, it should catch this instead, not Style/StringConcatenation.)

This doesn't actually apply in the example case, because the second line isn't a string literal. `Style/LineEndContinuation` wouldn't register in this case either.

That being said, `Layout/LineLength` can now break strings using continuations if you configure it with `SplitStrings: true`. There is a bug that would affect this case, but I have opened #13889 to fix it. Once merged, and with `SplitStrings: true`, the original example, as well as the "violates Style/StringConcatenation" example, will both autocorrect to:

```ruby
puts "This text ends with #{some_method}"
puts 'This text ends with ' \
     "#{some_much_longer_method}"
```
